app/services/ml.py

```diff
@@ -1,11 +1,99 @@
-import random
+from datetime import datetime, timedelta
+from typing import List
+
+import numpy as np
+import pandas as pd
+import tweepy
+from fastapi import Depends
+
+import model
+
+from .scrape import TwitterID, TwitterScraper
+
+
+def _find_replies_in_conversation(tweet_id: int, conversation: List[tweepy.Tweet]):
+    replies = []
+    for tweet in conversation:
+        for reference_tweet in tweet.referenced_tweets:
+            if reference_tweet.type == "replied_to" and reference_tweet.id == tweet_id:
+                replies.append(
+                    {
+                        "id": tweet.id,
+                        "text": tweet.text,
+                        "parent_id": tweet_id,
+                    }
+                )
+                break
+    return replies
 
 
 class ML:
+    def __init__(self, scraper: TwitterScraper = Depends()):
+        self._scraper = scraper
+        self._inference = model.Inference()
+
     def train(self):
         # TODO: for train / re-train purpose
         pass
 
-    def get_analysis_result(self, username: str):
-        # TODO: maybe ML model here
-        return round(random.uniform(0.0, 1.0), 2)
+    def get_analysis_result(self, username: str) -> float:
+        user_api = self._scraper.get_user_by_username(username)
+        user = self._make_ml_user(user_api)
+        tweets = self._get_ml_tweets(user_api.id)
+        return self._inference.predict(user, tweets)
+
+    def _get_ml_tweets(self, user_id: TwitterID):
+        tweets = []
+        duration = timedelta(days=7)
```

Comment:
Let's note here also
```suggestion
        # NOTE: can only fetch last 7 days unless have Academic Access or sth
        duration = timedelta(days=7)
```